index.d.ts

```diff
@@ -0,0 +1,197 @@
+import {KEYWORD} from 'color-convert/conversions';
+
+declare namespace ansiStyles {
+	interface ColorConvert {
+		/**
+		 * The RGB color space
+		 *
+		 * @param red (0-255)
+		 * @param green (0-255)
+		 * @param blue (0-255)
+		 */
+		rgb(red: number, green: number, blue: number): string;
+
+		/**
+		 * The HSL color space
+		 *
+		 * @param hue (0-360)
+		 * @param saturation (0-100)
+		 * @param lightness (0-100)
+		 */
+		hsl(hue: number, saturation: number, lightness: number): string;
+
+		/**
+		 * The HSV color space
+		 *
+		 * @param hue (0-360)
+		 * @param saturation (0-100)
+		 * @param value (0-100)
+		 */
+		hsv(hue: number, saturation: number, value: number): string;
+
+		/**
+		 * The HSV color space
+		 *
+		 * @param hue (0-360)
+		 * @param whiteness (0-100)
+		 * @param blackness (0-100)
+		 */
+		hwb(hue: number, whiteness: number, blackness: number): string;
+
+		/**
+		 * The CMYK color space
+		 *
+		 * @param cyan (0-100)
+		 * @param magenta (0-100)
+		 * @param yellow (0-100)
+		 * @param black (0-100)
+		 */
+		cmyk(cyan: number, magenta: number, yellow: number, black: number): string;
+
+		/**
+		 * The XYZ color space
+		 *
+		 * @param x (0-100)
+		 * @param y (0-100)
+		 * @param z (0-100)
+		 */
+		xyz(x: number, y: number, z: number): string;
+
+		/**
+		 * The LAB color space
+		 *
+		 * @param x (0-100)
+		 * @param y (0-100)
+		 * @param z (0-100)
+		 */
+		lab(lightness: number, a: number, b: number): string;
+
+		/**
+		 * The LCH/HCL color space
+		 *
+		 * @param lightness (0-100)
+		 * @param chroma (0-100)
+		 * @param hue (0-360)
+		 */
+		lch(lightness: number, chroma: number, hue: number): string;
+
+		/**
+		 * The RGB HEX color space
+		 *
+		 * @param hex A hexadecimal string containing RGB data
+		 */
+		hex(hex: string): string;
+
+		keyword(keyword: KEYWORD): string;
+		ansi(ansi: number): string;
+		ansi256(ansi: number): string;
+
+		/**
+		 * The HCG color space
+		 *
+		 * @param hue (0-360)
+		 * @param chroma (0-100)
+		 * @param gray (0-100)
+		 */
+		hcg(hue: number, chroma: number, gray: number): string;
+
+		/**
+		 * The Apple RGB 16-bit color space
+		 *
+		 * @param red (0-65535)
+		 * @param green (0-65535)
+		 * @param blue (0-65535)
+		 */
+		apple(r16: number, g16: number, b16: number): string;
+
+		/**
+		 * Grayscale color
+		 *
+		 * @param gray (0-100)
+		 */
+		gray(gray: number): string;
```

Comment:
Even though we just pass through the color modifiers to `color-convert`, I don't see the point in supporting all these conversions. Let's limit it to what we support in `chalk`: https://github.com/chalk/chalk#256-and-truecolor-color-support

Reply:
Well, the TypeScript definition is supposed to be accurate, and `ansi‑styles` currently supports all of the `color‑convert` color spaces, [which is a documented feature in `ansi‑styles`](https://github.com/chalk/ansi-styles#256--16-million-truecolor-support).

Removing support for those from `ansi‑styles` will be a breaking change,

Reply:
Yes, that's planned regardless: https://github.com/chalk/chalk/issues/360